### PR TITLE
ACM-8775: Global search merge relationships

### PR DESCRIPTION
**Pull request by jlpadilla**
### Related Issue
https://issues.redhat.com/browse/ACM-8775

### Description of changes
- Implement relationships for Global Search.

### CODE REVIEW

File: pkg/federated/mergeResponses.go [hunk 1]
```diff
@@ -86,24 +93,32 @@ func (d *Data) mergeMessages(msgs []string) {
...
+		return newItems
+	}
+
+	// Related results are grouped by kind.
+	for _, newItem := range newItems {
+		found := false
+		kind := newItem.Kind
+
+		// If the kind is found, merge the newItems with the mergedItems.
+		for index, mergedItem := range mergedItems {
```
Comment (SherinV) [inline pkg/federated/mergeResponses.go:109]: if the search term is not drilled down, like kind:Cluster, there might be a lot of mergedItems to loop through to find `kind`.

**Review by SherinV — APPROVED** [review]
Left one concern, but we can address it later, if an issue.
/lgtm
/hold

File: pkg/federated/mergeResponses.go (hunk 1 above)
Comment (jlpadilla) [reply pkg/federated/mergeResponses.go:109]: Agree, I'll merge this to get it working and revisit the performance after.

**Pull request merged**